File: src/tmechworks/TMechworks.scala
package tmechworks

import tmechworks.lib.Repo._
import cpw.mods.fml.common.{SidedProxy, FMLLog, Mod}
import cpw.mods.fml.common.network.NetworkMod
import tmechworks.network.PacketHandler
import cpw.mods.fml.common.event.{FMLPostInitializationEvent, FMLInitializationEvent, FMLPreInitializationEvent}
import cpw.mods.fml.common.Mod.EventHandler
import tmechworks.lib.ConfigCore
import net.minecraftforge.common.Configuration
import tmechworks.common.{MechContent, CommonProxy}
import tmechworks.common.CommonProxy
import net.minecraftforge.common.MinecraftForge

@Mod(modid = modId, name = modName, version = modVer, modLanguage = "scala")
@NetworkMod(serverSideRequired = false, clientSideRequired = true, channels = Array("TMechworks"), packetHandler = classOf[PacketHandler])
object TMechworks {

  @SidedProxy(clientSide = "tmechworks.client.ClientProxy", serverSide = "tmechworks.common.CommonProxy", modId = modId)
  var proxy: CommonProxy = null

  @EventHandler
  def preInit(evt:FMLPreInitializationEvent) {
    logger.setParent(FMLLog.getLogger)
    logger.info(s"$modName ($modVer) starting...")
    ConfigCore.loadConfig(new Configuration(evt.getSuggestedConfigurationFile))
    MechContent.setup()
    
    MinecraftForge.EVENT_BUS.register(new MultiblockEventHandler());
    proxy.registerTickHandler();
  }

  @EventHandler
  def init(evt:FMLInitializationEvent) {
        if (evt.getSide() == Side.CLIENT)
        {
            MinecraftForge.EVENT_BUS.register(new SignalTetherWorldOverlayRenderer());
        }
  }

  @EventHandler
  def postInit(evt:FMLPostInitializationEvent) {
    logger.info("Setup complete.")
  }

}

File: src/tmechworks/TMechworks.java
package tmechworks;

import java.util.logging.Logger;

import net.minecraftforge.common.Configuration;
import net.minecraftforge.common.MinecraftForge;
import tmechworks.client.SignalTetherWorldOverlayRenderer;
import tmechworks.common.CommonProxy;
import tmechworks.common.MechContent;
import tmechworks.lib.ConfigCore;
import tmechworks.lib.Repo;
import tmechworks.lib.TMechworksRegistry;
import tmechworks.lib.multiblock.MultiblockEventHandler;
import tmechworks.lib.util.TabTools;
import cpw.mods.fml.common.FMLCommonHandler;
import cpw.mods.fml.common.Mod;
import cpw.mods.fml.common.SidedProxy;
import cpw.mods.fml.common.Mod.EventHandler;
import cpw.mods.fml.common.Mod.Instance;
import cpw.mods.fml.common.event.FMLInitializationEvent;
import cpw.mods.fml.common.event.FMLPostInitializationEvent;
import cpw.mods.fml.common.event.FMLPreInitializationEvent;
import cpw.mods.fml.common.network.NetworkMod;
import cpw.mods.fml.relauncher.Side;

@Mod(modid = Repo.modId, name = Repo.modName, version = Repo.modVer, dependencies = "required-after:TConstruct")
@NetworkMod(serverSideRequired = false, clientSideRequired = true, channels = { "TMechworks" }, packetHandler = tmechworks.network.PacketHandler.class)
public class TMechworks {

    // Shared mod logger
    public static final Logger logger = Logger.getLogger("TMechworks");

    /* Instance of this mod, used for grabbing prototype fields */
    @Instance("TMechworks")
    public static TMechworks instance;
    /* Proxies for sides, used for graphics processing */
    @SidedProxy(clientSide = "tmechworks.client.ClientProxy", serverSide = "tmechworks.common.CommonProxy")
    public static CommonProxy proxy;

    public TMechworks ()
    {
        logger.setParent(FMLCommonHandler.instance().getFMLLogger());
    }

    @EventHandler
    public void preInit (FMLPreInitializationEvent event)
    {
        ConfigCore.loadConfig(new Configuration(event.getSuggestedConfigurationFile()));

        
        TMechworksRegistry.Mechworks = new TabTools("TMechworks");

        content = new MechContent();

        proxy.registerRenderer();
        proxy.registerTickHandler();
        
        MinecraftForge.EVENT_BUS.register(new MultiblockEventHandler());
    }

    @EventHandler
    public void init (FMLInitializationEvent event)
    {
        if (event.getSide() == Side.CLIENT)
        {
        	MinecraftForge.EVENT_BUS.register(new SignalTetherWorldOverlayRenderer());
        }
    }

    @EventHandler
    public void postInit (FMLPostInitializationEvent evt)
    {
        proxy.postInit();

    }

    public static MechContent content;
}
